clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -411,24 +411,21 @@ object GarbageCollector {
     val schema = StructType(Array(StructField("addresses", StringType, nullable = false)))
     val removed = {
       if (shouldSweep) {
-        // If a mark didn't happen in this run, gcAddressesLocation will be empty and expiredAddresses will be null.
-        if (gcAddressesLocation.isEmpty) {
+        // If a mark didn't happen in this run, gcAddressesLocation, runID, and expiredAddresses will be empty.
+        if (!shouldMark) {
           gcAddressesLocation = getAddressesLocation(storageNSForHadoopFS)
-        }
-        if (expiredAddresses == null) {
           expiredAddresses = readExpiredAddresses(gcAddressesLocation, markID)
+          runID = readRunIDFromMarkIDMetadata(gcAddressesLocation, markID)
```

Comment:
I would prefer to do this (only) when there is no runID already.

Also, I would like us to _log_ each time we computed one of these (i.e., on each `if` branch): I suspect it will make it easier to debug errors in whatever schedules the GC ruuns.

Reply:
`shouldMark == false` means that all of the three variables above are empty...

You would like to log that we fetched these values from a previous run?